firmware/Sources/Test/Test.c

```diff
@@ -0,0 +1,208 @@
+/*                     Definition of local symbolic constants                     */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Definition of local function like macros                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*            Definition of local types (typedef, enum, struct, union)            */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local variables                          */
+/**********************************************************************************/
+#ifdef EPC_CONF_PWM_ENABLED
+#endif
+
+#ifdef EPC_CONF_GPIO_ENABLED
+	#include "hal_gpio_test.h"
+#endif
+
+#ifdef EPC_CONF_ADC_DMA_ENABLED
+#endif
+
+#ifdef EPC_CONF_I2C_ENABLED
+
+#endif
+
+#ifdef EPC_CONF_TMR_ENABLED
+#endif
+
+#ifdef EPC_CONF_WDG_ENABLED
+#endif
+
+#ifdef EPC_CONF_CAN_ENABLED
+		HAL_CAN_result_e res;
+		uint8_t dataR = 0;
+		uint8_t sizeR = 0;
+		uint32_t idR = 0;
+#endif
+
+/**********************************************************************************/
+/*                        Definition of exported variables                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                      Definition of exported constant data                      */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                    Declaration of local function prototypes                    */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                       Definition of local constant data                        */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                         Definition of local functions                          */
+/**********************************************************************************/
+
+/**********************************************************************************/
+/*                        Definition of exported functions                        */
+/**********************************************************************************/
+
+TEST_result_e Test_PWM(void){
```

Comment:
All test defined in the same file?

Shouldnt they be defined in hal_gpio_test.c ...?

Reply:
Moved to test file all functions